pkg/driver/bacnet/merge/securityevent.go

```diff
@@ -0,0 +1,221 @@
+package merge
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"math"
+
+	"github.com/pborman/uuid"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/vanti-dev/gobacnet"
+	"github.com/vanti-dev/sc-bos/pkg/driver/bacnet/comm"
+	"github.com/vanti-dev/sc-bos/pkg/driver/bacnet/config"
+	"github.com/vanti-dev/sc-bos/pkg/driver/bacnet/known"
+	"github.com/vanti-dev/sc-bos/pkg/driver/bacnet/status"
+	"github.com/vanti-dev/sc-bos/pkg/gen"
+	"github.com/vanti-dev/sc-bos/pkg/gentrait/securityevent"
+	"github.com/vanti-dev/sc-bos/pkg/gentrait/statuspb"
+	"github.com/vanti-dev/sc-bos/pkg/node"
+	"github.com/vanti-dev/sc-bos/pkg/task"
+)
+
+type securityEventSource struct {
+	*config.ValueSource
+
+	Description string `json:"description"` // description of the security event, must be set to make sense of the event
+	// You want to set at least 1 of these OkBound values, else a security event will never get raised.
+	OkLowerBound *float64 `json:"okLowerBound,omitempty"` // if the point is equal to or greater than this value, it is ok.
+	OkUpperBound *float64 `json:"okUpperBound,omitempty"` // if the point is equal to or less than this value, it is ok.
+
+	Actor     *string `json:"actor,omitempty"`     // Optional. Actor of the security event, e.g. "John Doe"
+	EventType *int32  `json:"eventType,omitempty"` // Optional. the type of event, must be one of gen.SecurityEvent_EventType
+	Priority  *int32  `json:"priority,omitempty"`  // Optional. Priority of the security event, lower is more important
+	Source    *string `json:"source,omitempty"`    // Optional. Source of the security event, e.g. "Door 1"
```

Comment:
I'm not sure Actor makes any sense here. I can't think of a situation where a specific BACnet point would represent a specific actor interacting with a security system.